arbnode/batch_poster.go

```diff
@@ -1350,3 +1365,56 @@ func (b *BatchPoster) StopAndWait() {
 	b.dataPoster.StopAndWait()
 	b.redisLock.StopAndWait()
 }
+
+type BoolRing struct {
+	buffer         []bool
```

Comment:
Let's move this to arbutil and make it generic over T so it can be used in the future